src/json-patch-duplex.ts

```diff
@@ -524,6 +525,65 @@ module jsonpatch {
     return true;
   }
 
+  
+  /**
+     * Apply a single json-patch on an object tree
+     * Returns the result object.
+     */
+  export function applyPatch<T>(tree: T, patch: Patch<any>): T;
+  export function applyPatch<T>(tree: T, patch: Patch<any>, validate: boolean, touchOriginalTree: boolean): T;
+  export function applyPatch<T>(tree: T, patch: Patch<any>, validate = false, touchOriginalTree = true): T {
```

Comment:
I am not convinced we should have the third parameter, it bloats the signature and adds more branches to the function (you would have to test this for all operations). JSON Patch's sole purpose is to mutate objects. If you really want to clone the tree on every operation, it can be easily achieved through functional composition:
```ts
patches.reduce((obj, op) => applyOperation(deepClone(obj), op))
```